### PR TITLE
add configuration to generate MeasureReports in the background

### PR DESCRIPTION
**Pull request by roywanyaga**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Closes #[2307]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt [hunk 1]
```diff
@@ -101,6 +102,41 @@ constructor(
...
+          .parseTemplate(
+            bundleName = LocalizationHelper.STRINGS_BASE_BUNDLE_NAME,
+            locale = Locale.getDefault(),
+            template = it
+          )
+          .decodeJson<T>()
+      }
+      .filter {
+        it.configType.equals(configType.name, ignoreCase = true)
+      }*/
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:119]: can we remove this?

**Review by pld — COMMENTED** [review]
is this needed on 0.2.4?

**Comment by roywanyaga** [comment]
> is this needed on 0.2.4?

Yes, its needed for stock threshold calculations for eCBIS

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt (hunk 1 above)
Comment (roywanyaga) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt]: I have removed

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**